lang/test/org/partiql/lang/errors/ParserErrorsTest.kt

```diff
@@ -831,10 +1427,23 @@ class ParserErrorsTest : SqlParserTestBase() {
                 Property.COLUMN_NUMBER to 5L,
                 Property.TOKEN_TYPE to TokenType.STAR,
                 Property.TOKEN_VALUE to ion.newSymbol("*")
-            )
+            ),
+            targetParsers = setOf(ParserTypes.SQL_PARSER)
+        )
+        checkInputThrowingParserException(
+            "SUM(*)",
+            ErrorCode.PARSE_UNEXPECTED_TOKEN,
+            mapOf(
+                Property.LINE_NUMBER to 1L,
+                Property.COLUMN_NUMBER to 5L,
+                Property.TOKEN_TYPE to TokenType.STAR,
+                Property.TOKEN_VALUE to ion.newSymbol("*")
+            ),
+            targetParsers = setOf(ParserTypes.PARTIQL_PARSER)
         )
     }
 
+    // FIXME: PartiQL Parser says the error location is at the parenthesis, when the real error token is `*`
```

Comment:
Do we need to cut an issue for this?

Reply:
Yeah, Trino actually does something interesting (that I tried to replicate). ANTLR essentially throws exceptions at the last evaluated rule, but it doesn't throw the exception in the "deepest" rule. So, Trino, upon receiving a SyntaxError, creates some state machine to try to figure out the "deepest" rule to find the most precise location where the syntax error is really coming from.

I tried to replicate their state machine, but it didn't end up working for me. Decided to also just punt it because it seemed that it would add complexity to this PR when what we really need is to understand parse errors happening as a whole.

So, yeah, I'll create an issue and reference this and the Trino implementation. Linking soon.

Reply:
See #731